### DISCUSSION TITLE
Avoid redirecting in SPA

### DISCUSSION THREAD

**Discussion by kingyue737** [keycloak/keycloak#19629] [Q&A] [answered]
My team want to incorporate Keycloak to our spa. But whether it is to verify the login status or refresh the token, redirection to auth server is required, which means a complete reinitialization of SPA.

I want to avoid redirect. But according to [documentation](https://www.keycloak.org/docs/latest/securing_apps/#_modern_browsers), silent SSO check is not supported by modern browsers and will be deprecated in the future. Currently is there any way that I can verify login status or refresh token in SPA via Keycloak JS adapter without redirect?

**Comment by jonkoops** [comment, marked as answer]
Keycloak JS implements the OpenID Connect specification, just like the Keycloak server. Unfortunately even the specification admits that these features will cease working at some point.

We're exploring alternative solutions such as the [Federated Credential Management API](https://developer.mozilla.org/en-US/docs/Web/API/FedCM_API), but currently there is no way around this issue.

**Comment by inad9300** [comment]
Related: https://keycloak.discourse.group/t/js-adapter-in-spa-authenticate-in-background-without-redirect-silent-login-required/5400.
